src/TextField/TextField.js

```diff
@@ -345,9 +345,10 @@ class TextField extends Component {
   };
 
   handleInputChange = (event) => {
-    this.setState({hasValue: isValid(event.target.value)});
```

Comment:
That's a breaking change. Besides, I think that the controlled / uncontrolled behavior should be determined by the `value` property, not the `onChange` one.